flask_securest/authentication_providers/abstract_authentication_provider.py

```diff
@@ -24,9 +32,35 @@ class AbstractAuthenticationProvider(object):
     to return the username (or another uniquely identifying value)
     which will be logged
     """
-
     __metaclass__ = abc.ABCMeta
 
     @abc.abstractmethod
     def authenticate(self, userstore=None):
         raise NotImplementedError
+
+    @staticmethod
+    def retrieve_credentials_from_request():
```

Comment:
rename and move to utils. This method should called get_basic_http_authentication_info